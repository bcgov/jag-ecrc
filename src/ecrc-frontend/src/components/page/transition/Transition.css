.transition-content {
  flex-direction: column;
  padding-top: 50px;
  padding-bottom: 270px;
  margin-left: auto;
  margin-right: auto;
}
