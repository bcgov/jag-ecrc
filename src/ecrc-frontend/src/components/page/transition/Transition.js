/* eslint-disable react/jsx-one-expression-per-line */
import React, { useEffect } from "react";
import PropTypes from "prop-types";
import Header from "../../base/header/Header";
import Footer from "../../base/footer/Footer";
import "./Transition.css";

export default function Transition({
  page: { header, transitionReason = "bcsc" }
}) {
  // useEffect(() => {
  //   setTimeout(() => {
  //     sessionStorage.setItem("validExit", true);
  //     window.open("https://justice.gov.bc.ca/eCRC/home.htm", "_self");
  //   }, 6000);
  // }, []);

  return (
    <main>
      <Header header={header} />
      <div className="page">
        <div className="transition-content col-md-8">
          {transitionReason === "notwhitelisted" && (
            <p>
              We are transitioning our client organizations currently using
              Equifax for identity verification to a new process through a{" "}
              <a
                href="https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card"
                target="_blank"
                rel="noopener noreferrer"
              >
                BC Services Card
              </a>
              {". Your organization hasn't"} been transitioned to the new system
              yet.
            </p>
          )}
          {transitionReason === "bcsc" && (
            <div>
              <p>
                You are unable to login with your BC Services Card at this time.
              </p>
              <br />
              <p>
                We require users to have a photo BC Services Card to access the
                Criminal Record Check at this time.
              </p>
            </div>
          )}
          <br />
          <p>
            You will receive instructions on how to submit your criminal record
            check and will be required to re-enter your access code to continue.
            You will be redirected automatically. If you are not redirected
            automatically, please click{" "}
            <span
              data-testid="exitApp"
              className="pointer-here"
              onClick={() => {
                sessionStorage.setItem("validExit", true);
                window.open("https://justice.gov.bc.ca/eCRC/home.htm", "_self");
              }}
              role="button"
              onKeyDown={() => {
                sessionStorage.setItem("validExit", true);
                window.open("https://justice.gov.bc.ca/eCRC/home.htm", "_self");
              }}
              tabIndex={0}
            >
              here
            </span>
            {"."}
          </p>
        </div>
      </div>
      <Footer />
    </main>
  );
}

Transition.defaultProps = {
  page: {
    transitionReason: "bcsc"
  }
};

Transition.propTypes = {
  page: PropTypes.shape({
    header: PropTypes.shape({
      name: PropTypes.string.isRequired
    }).isRequired,
    transitionReason: PropTypes.string
  })
};
