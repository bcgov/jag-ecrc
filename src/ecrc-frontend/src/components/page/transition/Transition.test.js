import React from "react";
import { create } from "react-test-renderer";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { render, wait } from "@testing-library/react";

import Transition from "./Transition";

jest.setTimeout(7000);

describe("Transition Page Component", () => {
  const header = {
    name: "Criminal Record Check"
  };

  const page = {
    header
  };

  window.open = jest.fn();

  test("Matches the default/BCSC snapshot", () => {
    const history = createMemoryHistory();

    const defaultTransition = create(
      <Router history={history}>
        <Transition page={page} />
      </Router>
    );

    expect(defaultTransition.toJSON()).toMatchSnapshot();
  });

  test("Matches the non-whitelisted snapshot", () => {
    const history = createMemoryHistory();

    const nonWhitelistTransition = create(
      <Router history={history}>
        <Transition page={{ ...page, transitionReason: "notwhitelisted" }} />
      </Router>
    );
    expect(nonWhitelistTransition.toJSON()).toMatchSnapshot();
  });

  test("Component waits 6 seconds before automatically redirecting the user", async () => {
    const history = createMemoryHistory();

    render(
      <Router history={history}>
        <Transition page={{ ...page, transitionReason: "notwhitelisted" }} />
      </Router>
    );

    await wait(
      () => {
        expect(window.open).toHaveBeenCalled();
      },
      { timeout: 6000 }
    );
  }, 7000);
});
