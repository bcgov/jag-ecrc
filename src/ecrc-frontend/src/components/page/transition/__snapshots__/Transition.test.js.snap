// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transition Page Component Matches the default/BCSC snapshot 1`] = `
<main>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-none d-md-block"
            height="44"
            src="/images/bcid-logo-rev-en.svg"
            width="181"
          />
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-md-none"
            height="44"
            src="/images/bcid-symbol-rev.svg"
            width="64"
          />
        </a>
        <div>
          <a
            className="navbar-brand nav-item nav-link"
            href="/"
          >
            Criminal Record Check
          </a>
        </div>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        />
      </div>
    </nav>
  </header>
  <div
    className="page"
  >
    <div
      className="transition-content col-md-12"
    >
      <p>
        You are being redirected to the legacy online criminal record check (eCRC) application. You will be required to re-enter your access code to continue. If you are not redirected automatically, please click 
        <a
          href="https://justice.gov.bc.ca/eCRC/home.htm"
        >
          here
        </a>
        .
      </p>
      <br />
      <p>
        TEXT TO BE UPDATED You failed a bcsc login, did not have a bcsc with picture, did not have bcsc...etc
      </p>
    </div>
  </div>
  <footer
    className="footer"
  >
    <nav
      className="navbar navbar-expand-sm navbar-dark"
    >
      <div
        className="container"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disclaimer
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/privacy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              rel="noopener noreferrer"
              target="_blank"
            >
              Accessibility
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/copyright"
              rel="noopener noreferrer"
              target="_blank"
            >
              Copyright
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              rel="noopener noreferrer"
              target="_blank"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </footer>
</main>
`;

exports[`Transition Page Component Matches the non-whitelisted snapshot 1`] = `
<main>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-none d-md-block"
            height="44"
            src="/images/bcid-logo-rev-en.svg"
            width="181"
          />
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-md-none"
            height="44"
            src="/images/bcid-symbol-rev.svg"
            width="64"
          />
        </a>
        <div>
          <a
            className="navbar-brand nav-item nav-link"
            href="/"
          >
            Criminal Record Check
          </a>
        </div>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        />
      </div>
    </nav>
  </header>
  <div
    className="page"
  >
    <div
      className="transition-content col-md-12"
    >
      <p>
        You are being redirected to the legacy online criminal record check (eCRC) application. You will be required to re-enter your access code to continue. If you are not redirected automatically, please click 
        <a
          href="https://justice.gov.bc.ca/eCRC/home.htm"
        >
          here
        </a>
        .
      </p>
      <br />
      <p>
        We are transitioning our client organizations currently using Equifax for identity verification to a new process through a
         
        <a
          href="https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card"
        >
          BC Services Card account
        </a>
        . Your organization hasn't
         been transitioned to the new system yet.
      </p>
    </div>
  </div>
  <footer
    className="footer"
  >
    <nav
      className="navbar navbar-expand-sm navbar-dark"
    >
      <div
        className="container"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disclaimer
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/privacy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              rel="noopener noreferrer"
              target="_blank"
            >
              Accessibility
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/copyright"
              rel="noopener noreferrer"
              target="_blank"
            >
              Copyright
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              rel="noopener noreferrer"
              target="_blank"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </footer>
</main>
`;
